File: AN_2/SEM_2/MPP/java/lab4/src/main/java/ro/mpp/labfx/service/ParticipantCursaService.java
package ro.mpp.labfx.service;

import ro.mpp.labfx.domain.Cursa;
import ro.mpp.labfx.domain.Participant;
import ro.mpp.labfx.domain.ParticipantCursa;
import ro.mpp.labfx.repository.ParticipantCursaRepository;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class ParticipantCursaService {
    private ParticipantCursaRepository participantCursaRepository;

    public ParticipantCursaService(ParticipantCursaRepository participantCursaRepository) {
        this.participantCursaRepository = participantCursaRepository;
    }
    public int findNrPartByCursa(Cursa cursa){
        return participantCursaRepository.findNrPartByCursa(cursa);
    }

    public void adauga(Participant participant, Cursa cursa){
        participantCursaRepository.adaugainbaza(participant,cursa);
    }

    public void sterge(Participant participant, Cursa cursa){
        participantCursaRepository.sterge(participant.getId());
    }

    public List<ParticipantCursa> findAll(){
        return participantCursaRepository.findAll();
    }
}

File: AN_2/SEM_2/MPP/c#/lab4/lab4/service/ParticipantCursaService.cs
using lab4.domain;
using log4net;

namespace lab4.service;

public class ParticipantCursaService(IParticipantCursaRepository participantCursaRepo)
{
    private static readonly ILog Logger = LogManager.GetLogger(typeof(ParticipantCursaService));

    public int FindNrPartByCursa(Cursa cursa)
    {
        Logger.Debug("Incep cautarea participantilor pentru cursa.");
        
        if (cursa == null)
        {
            throw new ArgumentNullException(nameof(cursa), "Cursa cannot be null");
        }
        return participantCursaRepo.CountByCursa(cursa);
    }

    public void adaugainbaza(Participant participant, Cursa cursa)
    {
        Logger.Debug("Incep cautarea participantilor pentru cursa");
        participantCursaRepo.adaugainbaza(participant,cursa);
    }

    public List<ParticipantCursa> FindAll()
    {
        Logger.Debug("Incep cautarea participantilor pentru cursa");
        return participantCursaRepo.FindAll();
    }
}